src/rest/server/handler.go

```diff
@@ -302,3 +310,64 @@ func hostMetadataHandler(w http.ResponseWriter, r *http.Request) {
 	w.Header().Set("Content-Type", "application/xrd+xml")
 	w.Write(data.Bytes())
 }
+
+func Authenticate(w http.ResponseWriter, r *http.Request) {
+	var creds Credentials
+	err := json.NewDecoder(r.Body).Decode(&creds)
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	auth_success, err := UserPwAuth(creds.Username, creds.Password)
+	if  auth_success {
+		exp_tm := time.Now().Add(JwtValidInt)
+		token := generateJWT(creds.Username, exp_tm)
+		http.SetCookie(w, &http.Cookie{
```

Comment:
Is sending token in cookie a standard practice?

Reply:
There is no specific standard, it is done both ways, via cookie, or via Authorization header, we simply have to pick a method we prefer and go with it. Cookie vs Auth header both have some advantages/disadvantages, but nothing too major.